packages/grid/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -87,6 +87,10 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridRender
       [apiRef, props],
     );
 
+    if (rowNode.position === 'footer') {
```

Comment:
Note that this is not sufficient. The footer row is still selectable. One way is to add an `isRowSelectable` pre-processor.

Reply:
The concept of footer is known  by the community version so I think we can test it directly in the selection plugin without a pre-processor

Reply:
See https://github.com/mui/mui-x/pull/4208/commits/64093a0688b792d277c118e2013c47c538fdd04a